### PR TITLE
Canopy Cover Survey Model

### PR DESCRIPTION
**Pull request by athai**
fixes issue #62 

## Changes in this PR.

- [X] Write tests for Canopy Cover Survey
- [X] Write tests for CC_Cardinal (direction) boxes
- [x] Implement

## Testing this PR.

1. ``docker-compose run web bash``
2. ``cd`` streamwebs_frontend/
3. ``python manage.py test streamwebs/models/tests``

### Expected Output.

```
[root@4d703ed1cfdc streamwebs_frontend]# python manage.py test streamwebs/tests/models/
Creating test database for alias 'default'...
...............................
----------------------------------------------------------------------
Ran 31 tests in 0.506s

OK
Destroying test database for alias 'default'...

```

### CODE REVIEW

File: streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                                              west=west, est_canopy_cover=50)
+
+        self.assertEqual(canopyc.school, 'School A')
+        self.assertEqual(canopyc.date_time, default_dt)
+        self.assertEqual(canopyc.weather, 'cloudy')
+        self.assertEqual(canopyc.north.shaded, 11)
+        self.assertEqual(canopyc.east.shaded, 8)
+        self.assertEqual(canopyc.south.shaded, 14)
+        self.assertEqual(canopyc.west.shaded, 17)
+        self.assertEqual(canopyc.est_canopy_cover, 50)
```
Comment (athai) [inline streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py]: I considered adding assert statements to make sure that the boolean values were correctly assigned for each cardinal direction, but that will end up being very verbose... Do you think it's necessary to check for that?

For reference, it'll look something along the lines of:
```
self.assertEqual(canopyc.north.A, True)
self.assertEqual(canopyc.north.B, True)
...
self.assertEqual(canopyc.north.X, False)
```
for each cardinal box.

Comment (alxngyn) [inline streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py]: Maybe just test one cardinal direction?

**Comment by alxngyn** [comment]
+1 for the tests. There is probably some better 2D array implementation of this but I think it's fine for now.

File: streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py (hunk 1 above)
Comment (subnomo) [inline streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py]: If you want, you can create a list of all the True/False values like so:
```python
bools = [True, True, False, True...]
```
and use that to create the shades like so:

```python
north = CC_Cardinal.objects.create_shade(*bools, 11)
```

That way, you can [loop through each field](http://stackoverflow.com/questions/10353804/how-do-i-loop-through-fields-of-an-object) in `canopyc.north` and compare against that position in `bools`.

**Comment by subnomo** [comment]
Looks good. :+1:

File: streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                                                 False, True, False, True,
+                                                 True, False, True, False,
+                                                 False, 11)
+
+        east = CC_Cardinal.objects.create_shade('East', True, True, False,
+                                                False, False, True, False,
+                                                False, True, True, True, False,
+                                                True, False, False, False,
+                                                False, False, True, True,
+                                                False, False, False, False, 8)
```
Comment (leian7) [inline streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py]: It doesn't matter right now, but I think there are 9 Trues here instead of 8.

**Comment by leian7** [comment]
+1

File: streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py (hunk 2 above)
Comment (athai) [inline streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py]: Yep! I varied each cardinal direction so that they weren't the same.

**Pull request merged**